store/sqlstore/channel_store.go

```diff
@@ -955,14 +955,17 @@ func (s SqlChannelStore) GetChannels(teamId string, userId string, includeDelete
 			sq.And{
 				sq.Expr("Id = ChannelId"),
 				sq.Eq{"UserId": userId},
-				sq.Or{
-					sq.Eq{"TeamId": teamId},
-					sq.Eq{"TeamId": ""},
-				},
 			},
 		).
 		OrderBy("DisplayName")
 
+	if teamId != "" {
+		query = query.Where(sq.Or{
+			sq.Eq{"TeamId": teamId},
+			sq.Eq{"TeamId": ""},
+		})
+	}
```

Comment:
Squirrel Where is prepared so any further Where call will concatenate whatever it previously had with the new statement with an `AND`.

https://github.com/Masterminds/squirrel/blob/cd1fe0a38ba74020c727421ca469fc4a4d7f0258/select.go#L319-L321

This is needed for this PR, since we only want to search in certain team (and DM/GMS) when we are passing the team ID. If we are not passing a team ID, we want to look through all the posts.

Reply:
Okay I guess it was redundant before in the case of an empty team id.

Reply:
Before an empty team ID would have never reached this function.